test/Microsoft.Azure.Devices.Edge.Test/helpers/TestAuthenticationType.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Helpers
+{
+    using System.ComponentModel;
+    using Microsoft.Azure.Devices;
+
+    public enum TestAuthenticationType
+    {
+        SasInScope,
+        SasOutOfScope,
+        CertificateAuthority,
+        SelfSignedPrimary,
+        SelfSignedSecondary
+    }
+
+    public static class TestAuthenticationTypeExtensions
+    {
+        public static AuthenticationType ToAuthenticationType(this TestAuthenticationType authType)
+        {
+            switch (authType)
+            {
+                case TestAuthenticationType.SasInScope:
+                case TestAuthenticationType.SasOutOfScope:
+                    return AuthenticationType.Sas;
+                case TestAuthenticationType.CertificateAuthority:
+                    return AuthenticationType.CertificateAuthority;
+                case TestAuthenticationType.SelfSignedPrimary:
+                case TestAuthenticationType.SelfSignedSecondary:
+                    return AuthenticationType.SelfSigned;
+                default:
+                    throw new InvalidEnumArgumentException();
+            }
+        }
+
+        public static bool UseSecondaryCertificate(this TestAuthenticationType authType)
+        {
+            switch (authType)
```

Comment:
should we use a simple if statement?

Reply:
We could, but then the else clause would include invalid values (e.g. `(TestAuthenticationType)123456`). Not a big deal either way...I was just trying to be more precise. If we ever add another value to the enum, then these functions would throw until we add support for the new values.